### ISSUE TITLE
Is release v0.11.2 mistagged?

### ISSUE THREAD

**Issue by miccoli** [spack/spack#7292]
Reading the docs, I would assume that `master`, `releases/v0.11.2` and tag `v0.11.2` should point to the very same commit.

However 
```
$ git for-each-ref --format='%(refname:strip=2) %09 %(objectname:short)' refs/heads/master refs/heads/releases/v0.11.2 refs/tags/v0.11.2 | column -t
master            041aa143d
releases/v0.11.2  e905f8cf8
v0.11.2           8913c3465
```
While the differences between 041aa143d  and e905f8cf8 are trivial
```
git diff --stat 041aa143d e905f8cf8
 .travis.yml | 1 -
 1 file changed, 1 deletion(-)
```
8913c3465 (tag v0.11.2) is substantially different form the other two:
```
$ git diff --shortstat 8913c3465 e905f8cf8
 1033 files changed, 6262 insertions(+), 24101 deletions(-)
$ git diff --dirstat 8913c3465 e905f8cf8
  11.2% lib/spack/external/_pytest/
  34.1% lib/spack/external/jinja2/
   4.9% lib/spack/spack/
   3.8% lib/spack/
   3.7% var/spack/repos/builtin/packages/openfoam-com/
  41.5% var/spack/repos/builtin/packages/
```
Is tag v0.12.2 pointing to the wrong commit?

The same problem does not occur for tags v0.10.0, v0.11.0, v0.11.1 which point at the same commit of the respective releases branch.
More explicitly:
```
$ git diff --shortstat v0.11.1 origin/releases/v0.11.1
```
gives no differences, while:
```
$ git diff --shortstat v0.11.2 origin/releases/v0.11.2
 1033 files changed, 6262 insertions(+), 24101 deletions(-)
```

**Comment by tgamblin** [comment]
😱.  Yes.  It's misstaged.  It looks like I targeted `v0.11.2` at `develop` when it should've been pointed at `master`.  I need to script this better.

I think I will fix this by issuing a `0.11.3` release based on the releases/v0.11.2 branch, then just cutting a `0.12.0` based on current `develop`.  Would that work for you?

**Comment by tgamblin** [comment]
Closing this as I think by the time we got around to dealing with this, 0.12 was out.  We'll avoid mis-staged releases in the future -- sorry for the hassle!
